src/Cool/String/String.cpp

```diff
@@ -205,33 +258,73 @@ auto next_word(
     }
     else
     {
-        return std::string{text.substr(position->first, position->second - position->first)};
+        return substring(text, *position);
     }
 }
 
-auto next_block(
+auto find_block_position(
     std::string_view text,
-    size_t           ending_key_pos
-) -> std::optional<std::string>
+    size_t           offset
+) -> std::optional<std::pair<size_t, size_t>>
 {
-    const auto start_position_of_block = text.find("(") + 1;
-    if (start_position_of_block != std::string_view::npos)
+    const auto first_word_position  = find_next_word_position(text, offset);
+    auto       parentheses_position = find_matching_pair({.text = text, .offset = offset});
```

Comment:
Make it const. You can create a new pair when you need to return a modified version of parentheses_position